### PR TITLE
DAOS-6803 ofi: Update to ofi v1.12.0rc1

### PR DESCRIPTION
**Pull request by frostedcmos**
- Update to v1.12.0rc1

Signed-off-by: Alexander Oganezov <alexander.a.oganezov@intel.com>

### CODE REVIEW

File: libfabric.spec [hunk 1]
```diff
@@ -1,10 +1,8 @@
 %define suse_libname libfabric1
-#global dl_version 1.11.1rc1
-
-%{?dl_version:%global autosetup_args --p1 -n libfabric-%{dl_version}}
+%global dl_version 1.12.0rc1
```
Comment (brianjmurrell) [inline libfabric.spec:2]: We're starting to gravitate towards some standards that allow us to easily switch between _pre_ and _GA_ releases of packages without having to change lots of things all over the spec file.  First we start with defining the components of the version and build the `%dl_version` from it:

```suggestion
%global major 1
%global minor 12
%global bugrelease 0
%global prerelease rc1
%global dl_version  %{major}.%{minor}.%{bugrelease}%{?prerelease:%{prerelease}}
```

File: libfabric.spec [hunk 2]
```diff
@@ -1,10 +1,8 @@
...
 
 Name: libfabric
-Version: 1.11.1
+Version: 1.12.0~rc1
```
Comment (brianjmurrell) [inline libfabric.spec:5]: Next we define the version from the components:
```suggestion
Version: 1.12.0~rc1
Version: %{major}.%{minor}.%{bugrelease}%{?prerelease:~%{prerelease}}
```

File: libfabric.spec [hunk 3]
```diff
@@ -82,7 +80,7 @@ Requires: libpsm2-devel >= 11.2.78
 Development files for the libfabric library.
 
 %prep
-%autosetup %autosetup_args
+%autosetup -n libfabric-%dl_version
```
Comment (brianjmurrell) [inline libfabric.spec:83]: Then make a portable `%{autosetup}` that doesn't need changing every time:
```suggestion
%autosetup -p1 -n libfabric-%dl_version
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
GH review won't let me say so in the PR since it was not a line modified by you, but you also need to update https://github.com/daos-stack/libfabric/blob/1c5b13ddf7c356434f2fb45928f2b934dfff9747/libfabric.spec#L16 to
```
Source: https://github.com/ofiwg/%{name}/archive/v%{dl_version}.tar.gz
```

File: packaging/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/groovy
```
Comment (brianjmurrell) [inline packaging/Jenkinsfile:1]: This file doesn't need to be added.  Did you use `<path_to_local_packaging_dir>/install` to update the packaging?  IIRC, that only copied the needed files for a `packaging/` update.

File: packaging/Makefile [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+OSUSE_HPC_REPO := https://download.opensuse.org/repositories/science:/HPC
+LEAP_42_HPC_REPO := $(OSUSE_HPC_REPO)/openSUSE_Leap_42.3/
+
+ifeq ($(DAOS_STACK_LEAP_42_GROUP_REPO),)
+LEAP_42_REPOS  := $(LEAP_42_HPC_REPO)
+endif
+ifeq ($(DAOS_STACK_SLES_12_GROUP_REPO),)
+SLES_12_REPOS := $(LEAP_42_HPC_REPO)
+endif
```
Comment (brianjmurrell) [inline packaging/Makefile:13]: I think we can get rid of these lines now.  We soooo don't support those distros any more.

File: packaging/Makefile [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+ifeq ($(DAOS_STACK_LEAP_15_GROUP_REPO),)
+LEAP_15_REPOS := $(OSUSE_HPC_REPO)/openSUSE_Leap_15.1/
```
Comment (brianjmurrell) [inline packaging/Makefile:15]: This should probably be updated:
```suggestion
LEAP_15_REPOS := $(OSUSE_HPC_REPO)/openSUSE_Leap_15.2/
```

In fact it probably should be _determined automatically_:

```suggestion
LEAP_15_REPOS := $(OSUSE_HPC_REPO)/openSUSE_Leap_$(lsb_release -s -r)/
```

should work.

File: packaging/README.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# Common Packaging files and scripts
```
Comment (brianjmurrell) [inline packaging/README.md:1]: This file doesn't need to be added.

File: packaging/install [hunk 8]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (brianjmurrell) [inline packaging/install:1]: Nor this one.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Comment by frostedcmos** [comment]
Daos testing with new OFI passed in https://github.com/daos-stack/daos/pull/4666
Two failures there are known issues also happening on latest master: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/master/3872/

**Review by brianjmurrell — APPROVED** [review]

**Comment by brianjmurrell** [comment]
> Daos testing with new OFI passed in [daos-stack/daos#4666](https://github.com/daos-stack/daos/pull/4666)

Great.  Do you think we need to test this PR with current master to ensure that if we land it, it's compatible with everything that is in-flight or do you have confidence that there is nothing in this update that would break compatibility with current master?

**Comment by frostedcmos** [comment]
> > Daos testing with new OFI passed in [daos-stack/daos#4666](https://github.com/daos-stack/daos/pull/4666)
> 
> Great. Do you think we need to test this PR with current master to ensure that if we land it, it's compatible with everything that is in-flight or do you have confidence that there is nothing in this update that would break compatibility with current master?

I am not aware of anything that would break compatibility with this update. All OFI apis that mercury uses are backwards compatible and thats the only component that uses OFI directly.

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**